random.js

```diff
@@ -0,0 +1,76 @@
+var slideIndex = 0;
+var slides = document.getElementsByClassName("mySlides");
+var dots = document.getElementsByClassName("slide-inactive");
+var i;
+var userClick = false;
+let myTimeout;
+
+function currentSlide(n){
+    showSlides(slideIndex = n);
+}
+
+let showSlides = n => {
+    slideIndex > slides.length ? slideIndex = 1 : false;
+    n < 1 ? slideIndex = slides.length : false;
+
+    for(i = 0; i < slides.length; i++){
+        slides[i].style.display = "none";
+    }
+    for (i = 0; i < dots.length; i++) {
+        reAssign();
+    }
+    slides[slideIndex-1].style.display = "flex";
+    dots[slideIndex-1].className += " slide-active";
+}
+
+let slideAutomatic = () => {
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";  
+    }   
+
+    for (i = 0; i < dots.length; i++) {
+        reAssign();
+    }
+
+    slideIndex++;
+    slideIndex > slides.length ? slideIndex = 1 : false; 
+
+    slides[slideIndex-1].style.display = "flex";  
+    dots[slideIndex-1].className += " slide-active"; 
+
+    if(userClick == false){
+        myTimeout = setTimeout(slideAutomatic, 1000);
+    } 
+
+    else if(userClick == true){ 
+        stop();
+
+    } 
+    userClick = false;
+}
+
+function stop() { 
+    clearTimeout(myTimeout); 
+    setTimeout(slideAutomatic, 4000); 
+    console.log(userClick);
+    console.log("user has clicked."); 
+    userClick = false;
+} 
+
+document.addEventListener('click', function (event) {
```

Comment:
Also you can destructure the argument you want out of the event parameter. Here's an example

```
const foo = {
  x: '0',
  y: '1',
};
const { x } = foo;
console.log(x); // prints '0'
```

For your code 
```
function (event) {
```
It would be 

```
({ target })  => {
}
```

Which means you don't need `clickedElem` at all

Reply:
How could I implement that with my eventlistener click function

Reply:
Basically `event` (the argument to the eventListener function) is an object. Instead of taking that entire object just to only reference a single field `target` you can extract it directly in the signature and save a few lines.